### ISSUE TITLE
Examples documentation

### ISSUE THREAD

**Issue by superseed77** [kylemcdonald/ofxCv#157]
Is it possible to put one or two words to tell what each example is supposed to do ?
It's sometimes obvious, sometime not at all

**Comment by kylemcdonald** [comment]
you're totally right. i can't do this right now, but i would really appreciate if anyone has time to add some descriptions.

the best thing would be a comment in the testApp.cpp at the top of the file that looks like this:

```c++
/*
This example shows optical flow fields from a camera in realtime. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

More information about this technique is available at:
http://docs.opencv.org/master/d7/d8b/tutorial_py_lucas_kanade.html
Or in chapter 3 of the book OpenCV for Dummies.
*/
```

**Comment by avilleret** [comment]
wouldn't it be nice to also have a readme.md in the example folder ?
thus people can see it directly on github without the need to open the .cpp

**Comment by kylemcdonald** [comment]
@avilleret i like that idea. it would be best if it's one or the other, not both. i can see your adding some descriptions already in your fork. want to switch to adding the info to a `readme.md` (use lowercase filename) next to the project file? something like this:

```
# example-flow

This example shows optical flow fields from a camera in realtime. Lorem ipsum dolor sit amet,
consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
consequat.

More information about this technique is available in the [OpenCV documentation](http://docs.opencv.org/master/d7/d8b/tutorial_py_lucas_kanade.html), or in chapter 3 of the book OpenCV for Dummies.
```

**Comment by avilleret** [comment]
@kylemcdonald I don't konw which one from readme.md or comment in .cpp is the best. but I agree that one of them is enough. And I let you choose since it's your project after all :-)

**Comment by kylemcdonald** [comment]
i prefer having it in the `.cpp` to start with, and if it starts to get longer we can move it to a separate file.

i think most people using ofxCv will probably be opening the project files rather than just looking on github.

**Comment by pseudospencer** [comment]
Have started to do a bit of this myself as I'm going through the examples. Only problem is I'm not exactly sure what's going on in each. Posting what I have so far as a start and in case others want to add to this!

Flow: 
	• similar to flowtools. Direction vectors.

Threshold: 
	• Auto thresholds live video feed, converts to greyscale.

Edge: 
	• some really cool edge detection stuff that has thresholding as well

Difference: 
	• Extracts and displays the RGB difference between each frame + columns 

Difference: 
	• columns: same as difference but represents with a line/wave

AR: 
	• can't really tell, seems like I need a chessboard for this one

Background: 
	• Seems to be a background subtraction thing. Learns the background over time and only shows new pixels. Kind of keeps the old background for a burn-in effect

Bayer:
	• Seems to switch between image processing methods - BG2RGB, RG2RGB etc.
	• But starting with a b/w image that's already had some processing done on it

Blur:
	• Blurs an image from the camera feed. There's an option to use gaussian blur as well

Calibration:
	• Seems to be for calibrating to a checkerboard image or something? Unclear.

Calibration-LCP
	• Something about undistorting a distorted image from a camera that has a fisheye lens, etc.

Coherent lines
	• Seems like it does some additional processing on Canny edge detection to pick only lines that match with a threshold
	• Actually, nothing to do with canny edge detection - seems to be separate. But picks out big lines.

Contours-basic:
	• Seems to detect object areas. Contours based on brightness, shading?
	• Draws outline
	• Draws boxes for regions.
	• Adjustable threshold

**Comment by kaisark** [comment]
Hi. Anyone know where I can find the correct markers.png for the example-ar demo???
